src/renderer/mod.rs

```diff
@@ -59,11 +60,11 @@ pub struct Renderer {
     pub font_width: f32,
     pub font_height: f32,
     pub window_regions: Vec<WindowDrawDetails>,
-    pub batched_draw_command_receiver: Receiver<Vec<DrawCommand>>,
+    pub batched_draw_command_receiver: Option<Arc<Receiver<Vec<DrawCommand>>>>,
 }
 
 impl Renderer {
-    pub fn new(batched_draw_command_receiver: Receiver<Vec<DrawCommand>>) -> Renderer {

```

Comment:
Minor tidbit, but I think the default version of new should take the draw command receiver as I see the window as requiring it to function normally. But clearly you need a version of the renderer which doesn't have it hooked up to start, so I think we should have a separate new, maybe "new_unhooked" which constructs the renderer without the draw command channel? What do you think?